apps/judicial-system/backend/src/app/modules/notification/notification.service.ts

```diff
@@ -1224,39 +1245,67 @@ export class NotificationService {
   private async sendDefenderAssignedNotifications(
     theCase: Case,
   ): Promise<SendNotificationResponse> {
-    const defenderEmail = theCase.defenderEmail
-    if (!theCase.defenderNationalId || !defenderEmail) {
-      return Promise.resolve({ notificationSent: false })
-    }
-
+    const promises: Promise<Recipient>[] = []
     const pastNotifications = await this.notificationModel.findAll({
       where: { caseId: theCase.id, type: NotificationType.DEFENDER_ASSIGNED },
     })
-    const hasSentNotificationBefore = pastNotifications.some(({ recipients }) =>
-      recipients?.includes(defenderEmail),
-    )
 
-    if (hasSentNotificationBefore) {
-      return Promise.resolve({ notificationSent: false })
-    }
+    if (isIndictmentCase(theCase.type) && theCase.defendants) {
+      theCase.defendants.forEach((defendant) => {
+        const { defenderEmail, defenderNationalId, defenderName } = defendant
 
-    const { subject, body } = formatDefenderAssignedEmailNotification(
-      this.formatMessage,
-      theCase,
-      `${this.config.clientUrl}${DEFENDER_ROUTE}/${theCase.id}`,
-    )
+        if (!defenderNationalId || !defenderEmail) {
```

Comment:
Actually, I think the intention is to send a notification to a defender even if no national id is stored. This is done for ruling notifications. Then difference between defenders with a national id and without a national id is the contents of the email. A defender with national id will get a link to RVG, other defenders get an email and instructed to ask the court for documents.